### PR TITLE
add tenant support

### PR DESCRIPTION
**Pull request by Sathiyakugan**

### CODE REVIEW

**Comment by darshanasbg** [comment]
@ruwanta Agreed on the approach..
So neither user nor client (vs plugin code) have knowledge on a concept called tenant.

This will easily solve the handling oauth endpoints.

I prefer to access `/lsp` endpoint also accessed with the tenant qualified manner.. As we are requesting user to add base url with `http`, have to do minor additional processing to changint to `wss`. Hope this is ok?

File: client/src/Config.ts [hunk 1]
```diff
@@ -21,32 +21,33 @@
  */
 export class Config {
 
-    public static readonly WEBSOCKET_END_POINT = "wss://localhost:9443/lsp/debug";
+    public static readonly WEBSOCKET_END_POINT = "wss://localhost:9443/t/example.com/lsp/debug";
```
Comment (darshanasbg) [inline client/src/Config.ts:24]: As discussed, lets remove this

File: client/src/ui/oAuth.html [hunk 2]
```diff
@@ -13,123 +13,117 @@
...
-							<input id="clientSecret" type="password" class="form-control form-control-lg"
-								placeholder="Client Secret" required>
-						</div><br />
-						<div class="col-12 col-md-3 mx-auto">
-							<button id="login" type="submit" class="btn btn-block btn-lg btn-primary">Sign
-								In!</button>
-						</div>
-					</form>
-				</div>
+<!-- Masthead -->
```
Comment (jeradrutnam) [inline client/src/ui/oAuth.html:16]: Use 4 spaces for HTML indentation

File: client/src/DebugConstants.ts [hunk 3]
```diff
@@ -42,9 +42,8 @@ export class DebugConstants {
     public static readonly DEFAULT_SCRIPT_FILE = `defaultScriptFile`;
     public static readonly AUTHENTICATION_SCRIPT_TRUE = `AuthenticationScript enabled="true"`;
     public static readonly AUTHENTICATION_SCRIPT_FALSE = `AuthenticationScript enabled="false"`;
-    public static readonly IAM_URL = "IAM.URL";
+    public static readonly IAM_BASE_URL = "IAM.BaseURL";
```
Comment (brionmario) [inline client/src/DebugConstants.ts:45]: Type will be inferred from usage, but it's always best to declare the type.

File: client/src/Config.ts [hunk 4]
```diff
@@ -21,32 +21,32 @@
...
-    public static readonly PATH_APPLICATION_IMPORT: (iamUrl, tenant) => string =
-        (iamUrl, tenant) => `${iamUrl}/t/${tenant}/api/server/v1/applications/import`;
 
-    public static readonly PATH_GET_AUTH_CODE: (iamUrl, clientID, redirectUri, scope) => string =
-        (iamUrl, clientID, redirectUri, scope) => `${iamUrl}/oauth2/authorize?response_type=code&` +
-            `redirect_uri=${redirectUri}&client_id=${clientID}&scope="${scope}`;
 
-    public static readonly PATH_GET_SCRIPT_LIBRARY_BY_NAME: (iamUrl, tenant, scriptLibraryName) => string =
-        (iamUrl, tenant, scriptLibraryName) => `${iamUrl}/t/${tenant}/api/server/v1/` +
+    public static readonly PATH_APPLICATION_IMPORT: (baseUrl) => string =
```
Comment (thivi) [inline client/src/Config.ts:33]: Shall we specify the type of the argument here? For example: `(baseUrl: string) => string =`

File: client/src/ExtensionConstants.ts [hunk 5]
```diff
@@ -30,4 +30,8 @@ export class ExtensionConstants {
     public static readonly REFRESH_SERVICES = "extension.refreshServices";
     public static readonly REFRESH_SERVICES_VIEW_ID = "service-providers";
     public static readonly NODE_TLS_REJECT_UNAUTHORIZED = "NODE_TLS_REJECT_UNAUTHORIZED";
-}
\ No newline at end of file
+
+    public static readonly VS_CODE_MESSAGE_COMMAND_LOGIN= "login";
```
Comment (thivi) [inline client/src/ExtensionConstants.ts:34]: Let's add a space before the `=` sign.

File: client/src/ui/oAuth.html [hunk 6]
```diff
@@ -13,123 +13,117 @@
...
+						<input id="clientID" type="text" class="form-control form-control-lg"
+							   placeholder="Client ID" required>
+					</div><br />
+					<div class="col-md-12">
+						<input id="clientSecret" type="password" class="form-control form-control-lg"
+							   placeholder="Client Secret" required>
+					</div><br />
+					<div class="col-12 col-md-3 mx-auto">
+						<button id="login" type="submit" class="btn btn-block btn-lg btn-primary">Sign
+							In!</button>
```
Comment (thivi) [inline client/src/ui/oAuth.html:41]: I think it is better to move `Sign In!` to a new line between the tags.

File: client/src/DebugConstants.ts (hunk 3 above)
Comment (Sathiyakugan) [reply client/src/DebugConstants.ts:45]: https://github.com/wso2-extensions/identity-tools-plugin-vscode/pull/7

File: client/src/ExtensionConstants.ts (hunk 5 above)
Comment (Sathiyakugan) [reply client/src/ExtensionConstants.ts:34]: https://github.com/wso2-extensions/identity-tools-plugin-vscode/pull/7

File: client/src/ui/oAuth.html (hunk 6 above)
Comment (Sathiyakugan) [reply client/src/ui/oAuth.html:41]: https://github.com/wso2-extensions/identity-tools-plugin-vscode/pull/7

**Pull request merged**